legacy/src/main/java/com/amazon/opendistroforelasticsearch/sql/legacy/plugin/RestSqlAction.java

```diff
@@ -135,8 +142,19 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             LOG.info("[{}] Incoming request {}: {}", LogUtils.getRequestId(), request.uri(),
                     QueryDataAnonymizer.anonymizeData(sqlRequest.getSql()));
 
-            final QueryAction queryAction =
-                    explainRequest(client, sqlRequest, SqlRequestParam.getFormat(request.params()));
+            Format format = SqlRequestParam.getFormat(request.params());
+
+            // Route request to new query engine if it's supported already
+            SQLQueryRequest newSqlRequest = new SQLQueryRequest(sqlRequest.getJsonContent(),
+                                                                sqlRequest.getSql(),
+                                                                request.path(),
+                                                                format.getFormatName());
+            RestChannelConsumer result = newSqlQueryHandler.prepareRequest(newSqlRequest, client);
+            if (result != RestSQLQueryAction.NOT_SUPPORTED_YET) {
+                return result;
```

Comment:
add some log to tracking the fallback behaviour for debug

Reply:
sure, will do!